src/Hilke.KineticConvolution/Helpers/ShapeBuilder.cs

```diff
@@ -0,0 +1,149 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+using Fractions;
+
+using MoreLinq;
+
+namespace Hilke.KineticConvolution.Helpers
+{
+    internal class ShapeBuilder<TAlgebraicNumber>
```

Comment:
With this `ShapeBuilder`, there's no way to build two consecutive arcs, right?

Reply:
Yes: if the two arcs of consecutive corners meets at the same point, then no segment is constructed. 
For example:
Build a square of side `side` and round all four corners with `radius=0.5*side`. You obtain four arcs composing a circle.